controllers/update_pods.go

```diff
@@ -73,6 +74,19 @@ func (u UpdatePods) Reconcile(r *FoundationDBClusterReconciler, context ctx.Cont
 		}
 
 		if instance.Metadata.Annotations[LastSpecKey] != specHash {
+			if instance.GetProcessClass() == fdbtypes.ProcessClassStorage {
+				storageServersPerPodStr := getStorageServersPerPodForInstance(&instance)
+				storageServersPerPod, err := strconv.Atoi(storageServersPerPodStr)
+				if err != nil {
+					return false, err
+				}
+
+				// If the storage servers differ skip these instances since they will be replaced
+				if storageServersPerPod != cluster.GetStorageServersPerPod() {
```

Comment:
This check shouldn't be necessary, because we should be choosing pods for replacement before this step happens. We rely on that behavior in other areas, so that we don't have to duplicate logic between two actions.